### PR TITLE
Add stacking

### PR DESCRIPTION
**Pull request by gimoAI**
This PR adds:
- Option for stacking datasets
- Description of stacking to readme
- Tests for stacking

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -159,6 +159,34 @@ platform](https://github.com/asreview/systematic-review-datasets).
...
+#### Data format
+Your data files need to be in tabular file format or RIS file format. All input files should be in the same format, the output path should also be of the same file format.
+
+- **Tabular file format:**
+Supported tabular file formats are `.csv`, `.tab`, `.tsv` or `.xlsx`.
+
+
+- **RIS file format:**
+RIS file formats have `.ris` or `.txt` as extension and are used by digital libraries.
+Read [how to format](https://asreview.readthedocs.io/en/latest/data_format.html) your RIS files.
```
Comment (J535D165) [inline README.md:176]: In fact, it should be an ASReview-compatible format. Your tool will also work when someone creates a reader for another file format.

File: README.md [hunk 2]
```diff
@@ -159,6 +159,34 @@ platform](https://github.com/asreview/systematic-review-datasets).
 asreview data dedup benchmark:van_de_schoot_2017 -o van_de_schoot_2017_dedup.csv
 ```
 
+### Data Stack (Experimental)
```
Comment (J535D165) [inline README.md:162]: Vstack?

File: README.md [hunk 3]
```diff
@@ -159,6 +159,34 @@ platform](https://github.com/asreview/systematic-review-datasets).
...
+
+- **Tabular file format:**
+Supported tabular file formats are `.csv`, `.tab`, `.tsv` or `.xlsx`.
+
+
+- **RIS file format:**
+RIS file formats have `.ris` or `.txt` as extension and are used by digital libraries.
+Read [how to format](https://asreview.readthedocs.io/en/latest/data_format.html) your RIS files.
+
+#### Run script
```
Comment (J535D165) [inline README.md:178]:
```suggestion
```

Stay close to the format used in the other sections. If you disagree with the format, open another PR and change everything at once.

File: tests/test_stack.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+from pathlib import Path
+
+from asreview.data import ASReviewData
+from asreviewcontrib.datatools.stack import stack
+
+
+parent_dir = Path(__file__).parent
+file_1 = Path(parent_dir, "demo_data", "dataset_1.ris")
+file_2 = Path(parent_dir, "demo_data", "dataset_2.ris")
```
Comment (J535D165) [inline tests/test_stack.py:9]:
```suggestion
test_dir = Path(__file__).parent
file_1 = Path(test_dir, "demo_data", "dataset_1.ris")
file_2 = Path(test_dir, "demo_data", "dataset_2.ris")
```

**Review by J535D165 — CHANGES_REQUESTED** [review]
Nice, looks almost ready to merge.

File: README.md [hunk 5]
```diff
@@ -159,6 +159,26 @@ platform](https://github.com/asreview/systematic-review-datasets).
...
+
+Stack several datasets on top of each other: 
+```
+asreview data vstack output.csv MY_DATASET_1.csv MY_DATASET_2.csv MY_DATASET_3.csv
+```
+Here, 3 datasets are exported into a single dataset `output.csv`.
+The output path can be followed by any number of datasets to be stacked.
+
+#### Note
+Vstack does not do any deduplication.
```
Comment (Rensvandeschoot) [inline README.md:179]:
```suggestion
Vstack does not do any deduplication, and you might want to use the [deduplication ](https://github.com/asreview/asreview-datatools#data-dedup) script. 
```

File: README.md [hunk 6]
```diff
@@ -159,6 +159,26 @@ platform](https://github.com/asreview/systematic-review-datasets).
...
+If you wish to create a single (labeled, partly labeled, or unlabeled) dataset from multiple datasets while having control over duplicates and labels, use compose instead.
```
Comment (Rensvandeschoot) [inline README.md:180]:
```suggestion
If you wish to create a single (labeled, partly labeled, or unlabeled) dataset from multiple datasets containing labeling decisions while having control over duplicates, use [compose](https://github.com/asreview/asreview-datatools#data-compose) instead.
```

File: asreviewcontrib/datatools/stack.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    suffixes.append(Path(output_file).suffix)
+
+    set_ris = {".txt", ".ris"}
+    set_tabular = {".csv", ".tab", ".tsv", ".xlsx"}
+    set_suffixes = set(suffixes)
+
+    if len(set(suffixes)) > 1:
+        if not (set_suffixes.issubset(set_ris) or set_suffixes.issubset(set_tabular)):
+            raise ValueError(
+                "Several file types were given, all input files as well as the output file should be of the same type."
```
Comment (Rensvandeschoot) [inline asreviewcontrib/datatools/stack.py:21]:
```suggestion
                "•	Several file types were given; All input files, as well as the output file, should be of the same type."
```

**Review by Rensvandeschoot — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (gimoAI) [reply README.md:180]: added this, although reference will only work when merged with PR #19

**Review by J535D165 — APPROVED** [review]

**Comment by J535D165** [comment]
Thanks, I marked this PR as finished and will merge it just before the new release of datatools (because of the readme update).

**Pull request merged**